src/phoenix/datasets/evaluators.py

```diff
@@ -61,6 +67,150 @@ def evaluate(self, example: Example, exp_run: ExperimentRun) -> EvaluationResult
+        if name is not None:
+            self.name = name
+        else:
+            name = "LLMCriteriaEvaluator"
+
+    def evaluate(self, example: Example, exp_run: ExperimentRun) -> EvaluationResult:
+        assert exp_run.output is not None
+        result = _unwrap_json(exp_run.output.result)
+        formatted_template = self.template.replace("{text}", str(result))
+        unparsed_response = self.model._generate(formatted_template)
+        raw_label, explanation = (
+            self._parse_label_from_explanation(unparsed_response),
+            unparsed_response,
+        )
+        label = snap_to_rail(raw_label, ["true", "false"])
+        if label == "true":
+            score = 1.0
+        elif label == "false":
+            score = 0.0
+        else:
+            score = None
+        return EvaluationResult(
+            score=score,
+            explanation=explanation,
+            metadata={},
+        )
+
+    async def async_evaluate(self, example: Example, exp_run: ExperimentRun) -> EvaluationResult:
+        assert exp_run.output is not None
+        result = _unwrap_json(exp_run.output.result)
+        formatted_template = self.template.replace("{text}", str(result))
+        unparsed_response = await self.model._async_generate(formatted_template)
+        raw_label, explanation = (
+            self._parse_label_from_explanation(unparsed_response),
+            unparsed_response,
+        )
+        label = snap_to_rail(raw_label, ["true", "false"])
+        if label == "true":
+            score = 1.0
+        elif label == "false":
+            score = 0.0
+        else:
+            score = None
+        return EvaluationResult(
+            score=score,
+            explanation=explanation,
+            metadata={},
+        )
+
+    def _parse_label_from_explanation(self, raw_string: str) -> str:
+        label_delimiter = r"(\W*label\W*)"
+        parts = re.split(label_delimiter, raw_string, flags=re.IGNORECASE)
+        if len(parts) > 1:
+            # Find the last occurrence of the delimiter and take the part after it
+            last_index = len(parts) - 1
+            while last_index > 0:
+                if re.match(label_delimiter, parts[last_index - 1], flags=re.IGNORECASE):
+                    return parts[last_index].strip()
+                last_index -= 1
+        return raw_string
+
+    @classmethod
+    def _format_base_template(cls, criteria: str, description: Optional[str] = None) -> str:
+        if description is not None:
+            formatted_description = cls._description.replace("{criteria}", criteria)
+            formatted_description = formatted_description.replace("{description}", str(description))
+        else:
+            formatted_description = ""
+        formatted_template = cls._base_template.replace("{criteria}", str(criteria))
+        formatted_template = formatted_template.replace("{description}", str(formatted_description))
+        return formatted_template
```

Comment:
Bit confused why this is not a generic template application?

Reply:
what do you mean by generic template application? I think this is pretty standard